plugin/pkg/admission/podtolerationrestriction/admission.go

```diff
@@ -139,7 +139,7 @@ func (p *podTolerationsPlugin) Admit(a admission.Attributes) error {
 	}
 	pod.Spec.Tolerations = finalTolerations
 
-	return p.Validate(a)
```

Comment:
same here, this factoring was intentional. think of the validate phase as a double check that what was done/checked in the Admit phase still holds